swim/test_utils.go

```diff
@@ -121,6 +122,44 @@ func bootstrapNodes(t *testing.T, testNodes ...*testNode) {
 		})
 		require.NoError(t, err, "node must bootstrap successfully")
 	}
+	return hostports
+}
+
+func waitForConvergence(t *testing.T, timeout time.Duration, testNodes ...*testNode) {
```

Comment:
I don't think that will change anything here except for the test taking longer. We call `ProtocolPeriod` every iteration so we can move as fast as this method allows us. It blocks till the ping response is received.